File: src/prod/src/managed/Api/src/System/Fabric/Management/fabricdeployer/FlatNetwork/NetworkConfig.cs
// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
// ------------------------------------------------------------

namespace System.Fabric.FabricDeployer
{
    public class NetworkConfig
    {
        public string Name { get; set; }
        public string Id { get; set; }
        public string Driver { get; set; }
        public bool CheckDuplicate { get; set; }
        public IpamConfig Ipam { get; set; }
        public dynamic Options { get; set; }
    }
}

File: src/prod/src/Hosting2/NetworkConfig.cpp
// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
// ------------------------------------------------------------

#include "stdafx.h"

using namespace Hosting2;
using namespace Common;

Common::WStringLiteral const NetworkConfig::NetworkName(L"Name");
Common::WStringLiteral const NetworkConfig::Driver(L"Driver");
Common::WStringLiteral const NetworkConfig::CheckDuplicate(L"CheckDuplicate");
Common::WStringLiteral const NetworkConfig::NetworkIPAM(L"Ipam");

NetworkConfig::NetworkConfig()
{
}

NetworkConfig::NetworkConfig(
    std::wstring name,
    std::wstring driver,
    bool checkDuplicate,
    Hosting2::NetworkIPAM networkIPAM) :
    name_(name),
    driver_(driver),
    checkDuplicate_(checkDuplicate),
    networkIPAM_(networkIPAM)
{
}
